### ISSUE TITLE
Should split ffi into its own secp256k1-sys crate

### ISSUE THREAD

**Issue by apoelstra** [rust-bitcoin/rust-secp256k1#58]
The reason for this is that if users try to compile multiple versions of the FFI bindings at once they'll get link errors. Since most of our development happens in non-FFI areas it would be better to have a -sys crate that rarely updates while the main crate can do its own thing.

The transition from 0.10 to 0.11 was more painful than it needed to be because every crate depending on rust-secp had to update at once (since trying to compile crates with mismatching version would be a link error).

Doing this will be a FFI break in itself, of course, but we're planning to do this in the next version anyway to support free no-capability contexts.

**Comment by TheBlueMatt** [comment]
Does this really help the issue? I guess its a question of the rate of update of libsecp256k1 vs the rust bindings, but hopefully the rust bindings will calm down a bit going forward.

**Comment by apoelstra** [comment]
Yeah, maybe after this next batch of changes the ffi bindings aren't going to change at all without upstream impetus.
